x-pack/legacy/plugins/lens/public/editor_frame_plugin/editor_frame/workspace_panel.tsx

```diff
@@ -178,46 +181,70 @@ export function InnerWorkspacePanel({
   function renderVisualization() {
     useEffect(() => {
       // reset expression error if component attempts to run it again
-      if (expressionError) {
-        setExpressionError(undefined);
+      if (expression && localState.expressionBuildError) {
+        setLocalState(s => ({
+          ...s,
+          expressionBuildError: undefined,
+        }));
       }
     }, [expression]);
 
     if (expression === null) {
       return renderEmptyWorkspace();
     }
 
-    if (expressionError) {
+    if (localState.expressionBuildError) {
       return (
-        <EuiFlexGroup direction="column">
+        <EuiFlexGroup direction="column" className="lns">
```

Comment:
This is an incomplete class, and to get it centered add `alignItems`
```suggestion
        <EuiFlexGroup direction="column" alignItems="center">
```